drivers/hwinfo/hwinfo_gecko.c

```diff
@@ -139,6 +149,9 @@ int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
 
 #endif /* defined(BU_PRESENT) && defined(_SILICON_LABS_32B_SERIES_0) */
 
+	/* To fix compiler warning when none of the above flags are defined*/
+	rmu_flags = 0;
+
```

Comment:
These 3 lines are redundant and should be removed. rmu_flags is always initialized in line 38 and rmu_flags is never read after this line.

Reply:
Got a compilation warning as below:
C:/work/zephyrproject/zephyr/drivers/hwinfo/hwinfo_gecko.c:38:11: warning: unused variable 'rmu_flags' [-Wunused-variable]
   38 |  uint32_t rmu_flags = RMU_ResetCauseGet();
      |           ^~~~~~~~~

Reply:
OK, in that case using `ARG_UNUSED(rmu_flags)` is probably better. Just setting a variable without reading it may also create compiler warnings or issues with static code analysis tools.